lib/socket.js

```diff
@@ -551,11 +588,33 @@ internals.Socket.prototype._setCredentials = function (credentials, artifacts) {
     this.auth.isAuthenticated = true;
     this.auth.credentials = credentials;
     this.auth.artifacts = artifacts;
+    this.auth._verified = Date.now();
 
     this._listener._sockets.auth(this);
 };
 
 
+internals.Socket.prototype._verifyAuth = function () {
+
+    if (!this.auth.isAuthenticated) {
+        // no credentials available - no need to verify them
+        return true;
+    }
+
+    if (Date.now() - this.auth._verified < this._listener._settings.auth.verify) {
```

Comment:
I'm not adding a way to disable this completely - should I? By e.g. setting `verify: false`?

Reply:
Sounds reasonable. `false` works for me.